course_catalog/models.py

```diff
@@ -74,12 +88,105 @@ class Course(TimestampedModel):
+    language = models.CharField(max_length=128, null=True, blank=True)
+    year = models.IntegerField(null=True, blank=True)
+    full_description = models.TextField(null=True, blank=True)
+    start_date = models.DateTimeField(null=True, blank=True)
+    end_date = models.DateTimeField(null=True, blank=True)
+    enrollment_start = models.DateTimeField(null=True, blank=True)
+    enrollment_end = models.DateTimeField(null=True, blank=True)
+    image_src = models.URLField(max_length=400, null=True, blank=True)
+    image_description = models.CharField(max_length=1024, null=True, blank=True)
+    last_modified = models.DateTimeField(null=True, blank=True)
+    featured = models.BooleanField(default=False)
+    published = models.BooleanField(default=True)
+    availability = models.CharField(max_length=128, null=True, blank=True)
+    url = models.URLField(null=True, max_length=2048)
+    instructors = models.ManyToManyField(CourseInstructor, blank=True)
+    learning_resource_type = models.CharField(
+        max_length=20, default=ResourceType.course.value
+    )
+    prices = models.ManyToManyField(CoursePrice, blank=True)
+    location = models.CharField(max_length=128, null=True, blank=True)
+
+
+class List(LearningResource):
+    """
+    List model tracks an ordered list of other LearningResources.
+    """
+
+    image_src = models.URLField(max_length=400, null=True, blank=True)
+    image_description = models.CharField(max_length=1024, null=True, blank=True)
+
+    class Meta:
+        abstract = True
+
+
+class ListItem(TimestampedModel):
+    """
+    ListItem model tracks associated metadata and LearningResource.
+    `content_type` is restricted to the learning resources we want.
+    Lists should not contain other Lists (Programs, LearningPaths).
+    """
+
+    position = models.PositiveIntegerField()
+    content_type = models.ForeignKey(
+        ContentType,
+        limit_choices_to={"model__in": ("Course", "Bootcamp")},
+        on_delete=models.CASCADE,
+    )
+    object_id = models.PositiveIntegerField()
+    item = GenericForeignKey("content_type", "object_id")
+
+    class Meta:
+        abstract = True
+
+
+class LearningPath(List):
+    """
+    LearningPath is a user-created model tracking a restricted list of LearningResources.
+    """
+
+    author = models.ForeignKey(User, on_delete=models.PROTECT)
+    privacy_level = models.CharField(max_length=32, null=True, blank=True)
+
+
+class LearningPathItem(ListItem):
+    """
+    ListItem model for LearningPaths
+    """
+
+    list = models.ForeignKey(
+        LearningPath, related_name="items", on_delete=models.CASCADE
+    )
+
+
+class Program(List):
+    """
+    Program model for MIT programs. Consists of specified list of LearningResources.
+    """
+
+    pass
+
+
+class ProgramItem(ListItem):
+    """
+    ListItem model for Programs
+    """
+
+    list = models.ForeignKey(Program, related_name="items", on_delete=models.CASCADE)
```

Comment:
Should rename this field to not be a python builtin.

Reply:
I had written down the same concern in the spec! Any ideas for proposed names?

Reply:
I've changed this from `list` to `program` for now